### ISSUE TITLE
Limit gallery of images display

### ISSUE THREAD

**Issue by Infolu** [rilwis/meta-box#876]
Staff wordpress default gallery to limit the number of images to display. with args numberpost

Example
`<?php
	$images = get_children( array( 'post_parent' => $post->ID, 'post_type' => 'attachment', 'post_mime_type' => 'image', 'orderby' => 'menu_order', 'order' => 'ASC', 'numberposts' => 5 ) );
	if ( $images ) :
		$total_images = count( $images );
		$image = array_shift( $images );
		$image_img_tag = wp_get_attachment_image( $image->ID, 'thumbnail' );
?>
<figure class="gallery-thumb">
	<a href="<?php the_permalink(); ?>"><?php echo $image_img_tag; ?></a>
</figure>
<?php endif; ?>`

Now how would I do this with metabox? I tried this

`$images = rwmb_meta( 'media_photos_img', 'type=image&size=YOURSIZE&numberposts=1' ); // Prior to 4.8.0`

however for me it did not work.
Basically I want to display only the first image of an upload box for more than that box has 10 pictures.

**Comment by rilwis** [comment]
Just add `limit` param to the helper function for both file and image fields, please try:

```php
$images = rwmb_meta( 'media_photos_img', 'size=SIZE&limit=1' );
```

**Comment by Infolu** [comment]
Hello Rilwis, thanks for always answer, I as follows below however worked did something wrong? it keeps bringing me all images instead of a single image

`$images = rwmb_meta( 'media_gallery_img', 'size=SIZE&limit=1' ); // Prior to 4.8.0
if ( !empty( $images ) ) {
    foreach ( $images as $image ) {
        echo "<a href='{$image['full_url']}' rel='lightbox'><img src='{$image['url']}' width='{$image['width']}' height='{$image['height']}' alt='{$image['alt']}' /></a>";
    }
}
    ?>`

**Comment by funkedgeek** [comment]
He just added the update, so update the plugin from GitHub

**Comment by Infolu** [comment]
Hi @funkedgeek,  I saw that was added to the file file.php that limit could be added to type: image, type: image_advanced, type: plupload_image, type: image_upload, type: thickbox_image.

**Comment by rilwis** [comment]
It's inherited from file to all image files ;)

**Comment by Infolu** [comment]
Hello @rilwis  unfortunately for me it did not work, even using the latest version of metabox
follows the code I'm using, I still can not limit the display
`<?php
$images = rwmb_meta( 'media_gallery_img', 'type=image&size=YOURSIZE&limit=1' ); // Prior to 4.8.0
if ( !empty( $images ) ) {
    foreach ( $images as $image ) {
        echo "<a href='{$image['full_url']}' rel='lightbox'><img src='{$image['url']}' width='{$image['width']}' height='{$image['height']}' alt='{$image['alt']}' /></a>";
    }
}
 ?>`

**Comment by rilwis** [comment]
Just fixed that. Please try again :)

**Comment by Infolu** [comment]
Good, working perfectly, thanks
